api/v1alpha1/templates_common.go

```diff
@@ -30,27 +31,33 @@ const (
 	ChartAnnotationProviderName = "cluster.x-k8s.io/provider"
 
 	chartAnnoCAPIPrefix = "cluster.x-k8s.io/"
+
+	DefaultRepoName = "hmc-templates"
 )
 
-// +kubebuilder:validation:XValidation:rule="(has(self.chartName) && !has(self.chartRef)) || (!has(self.chartName) && has(self.chartRef))", message="either chartName or chartRef must be set"
+var DefaultSourceRef = sourcev1.LocalHelmChartSourceReference{
+	Kind: sourcev1.HelmRepositoryKind,
+	Name: DefaultRepoName,
+}
+
+// +kubebuilder:validation:XValidation:rule="(has(self.chart) && !has(self.chartRef)) || (!has(self.chart) && has(self.chartRef))", message="either chart or chartRef must be set"
 
 // HelmSpec references a Helm chart representing the HMC template
 type HelmSpec struct {
+	// Chart defines the desired state of the HelmChart to be created by the controller
+	Chart *sourcev1.HelmChartSpec `json:"chart,omitempty"`
```

Comment:
Yes. It'll be much better. I think we should do it. Thanks.

Reply:
It seems I spoke too soon. Embedding the `HelmChartSpec` struct causes issues with the current Kubebuilder validation for ensuring mutual exclusivity between the `ChartRef` and `HelmChartSpec` subfields. The `HelmChartSpec` struct includes required fields (`chart`, `sourceRef`, etc) that won’t be populated when `ChartRef` is set, leading to validation errors. To address this, I’ve renamed the `Chart` field to `ChartSpec` to avoid redundancy and prevent a `helm.chart.chart` naming issue.